### PR TITLE
Enhancement/create util function extract options

### LINKED ISSUE #8: Create util function: extractOptions
that extracts options/suggestion from data

https://github.com/zeljko-djuric/adverity-coding-challenge/blob/master/src/components/DisplayMetrics.js#L16-L23

relates to #2 

- [x] put into a separate folder
- [x] write tests
- [x] rename `*suggestions` to `*options` across the codebase
- [x] use result in `setState` of https://github.com/zeljko-djuric/adverity-coding-challenge/blob/master/src/components/DisplayMetrics.js#L15

### PR DESCRIPTION
**Pull request by zeljko-djuric**
Fixes #8

### CODE REVIEW

File: src/components/DisplayMetrics.js [hunk 1]
```diff
@@ -1,32 +1,26 @@
 import React from "react";
 import Select from "react-select";
 import _ from "lodash";
-
+import extractOptions from "../utils/extractOptions";
```
Comment (vlabs) [inline src/components/DisplayMetrics.js:4]: a empty line between imports and `class` is recommend

File: src/components/DisplayMetrics.js [hunk 2]
```diff
@@ -1,32 +1,26 @@
...
 class DisplayMetrics extends React.Component {
   constructor(args) {
     super(args);
     this.state = {
-      suggestions: [],
+      options: [],
```
Comment (vlabs) [inline src/components/DisplayMetrics.js:11]: :+1:

File: src/utils/extractOptions.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const extractOptions = data => {
+  var options = [];
```
Comment (vlabs) [inline src/utils/extractOptions.js]: avoid `var` or `const` by any means

File: src/utils/extractOptions.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  data.data.forEach((element, i) => {
```
Comment (vlabs) [inline src/utils/extractOptions.js]: also `forEach` should be avoided

try a more functional approach with `map`

**Review by vlabs — CHANGES_REQUESTED** [review]

File: src/utils/extractOptions.js (hunk 3 above)
Comment (vlabs) [reply src/utils/extractOptions.js]: sorry :)

it should be 
avoid `var` or `let` by any means

so always use `const`

Comment (zeljko-djuric) [reply src/utils/extractOptions.js]: Yea, I was a little bit confused after reading an article and your comment but now I am cool. :)
bc288c347ed2e128924dbbfaf66f355d8cf9daac

**Comment by vlabs** [comment]
please answer change rquest with commit hashes so i know what has been addressed

File: src/utils/extractOptions.js [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
-  data.data.map(item => {
-    let campaign = {};
-    let channel = {};
+  const options = [];
+  _.flatMap(data.data, item => {
+    const campaign = {};
+    const channel = {};
     campaign["label"] = item.campaign;
     channel["label"] = item.channel;
     options.push(campaign);
```
Comment (vlabs) [inline src/utils/extractOptions.js:10]: no need to `push`, just return it, see docs

**Review by vlabs — CHANGES_REQUESTED** [review]

File: src/utils/extractOptions.js (hunk 4 above)
Comment (zeljko-djuric) [reply src/utils/extractOptions.js]: 7f301a73a71fe172d902906517edf5f97374d9d7

**Comment by vlabs** [comment]
conflicts!

File: src/components/DisplayMetrics.js (hunk 1 above)
Comment (vlabs) [reply src/components/DisplayMetrics.js:4]: resolved

File: src/utils/extractOptions.js (hunk 5 above)
Comment (vlabs) [reply src/utils/extractOptions.js:10]: please resolve

**Comment by zeljko-djuric** [comment]
#13

File: src/utils/extractOptions.js (hunk 5 above)
Comment (zeljko-djuric) [reply src/utils/extractOptions.js:10]: e3930bd24c19058065b931c37cb091594f1b1acb

**Pull request closed without merging**